tiledb/sm/c_api/tiledb.cc

```diff
@@ -4801,6 +4801,38 @@ int32_t tiledb_array_evolve(
   return TILEDB_OK;
 }
 
+int32_t tiledb_array_upgrade_version(tiledb_ctx_t* ctx, const char* array_uri) {
+  // Sanity Checks
+  if (sanity_check(ctx) == TILEDB_ERR)
+    return TILEDB_ERR;
+
+  // Check array name
+  tiledb::sm::URI uri(array_uri);
+  if (uri.is_invalid()) {
+    auto st = Status::Error("Failed to find the array; Invalid array URI");
+    LOG_STATUS(st);
+    save_error(ctx, st);
+    return TILEDB_ERR;
+  }
+
+  // Create key
+  tiledb::sm::EncryptionKey key;
```

Comment:
This should support upgrading encrypted arrays to. Inside the storage manager class we should use the config to check if the key is set like we do for the newer open array API that has been merged in. We likely need to add a config parameter to the API here to support the key.

Reply:
added config* parameter. thanks